### ISSUE TITLE
Capturing a refinery while a harvester starts docking unloads cargo into capturing player's bank

### ISSUE THREAD

**Issue by abcdefg30** [OpenRA/OpenRA#18835]
The ownership of the harvester does not change. See https://github.com/OpenRA/OpenRA/pull/18737#pullrequestreview-530779141.

**Comment by pchote** [comment]
Do we want ownership to change in this case, or should the harvester immediately undock instead?

**Comment by OmegaBolt** [comment]
That should ideally be an option IMO. Definitely the case in TD through to RA2 even that harvesters ownership transfered if the building was captured while they are docked (not just when starting to dock). This affected TD and TS the most since harvesters spend forever in the refinery.

Ownership of vehicles that have just been produced at, but not yet started to leave, the war factory were also transfered.

**Comment by pchote** [comment]
We do already (and always have) support transferring ownership when the harvester is already docked. This (unless i'm mistaken) covers the time when the harvester is dragging in, which you could argue either way as to whether it should count or not. The turn before dragging certainly shouldn't count.

**Comment by abcdefg30** [comment]
> This (unless i'm mistaken) covers the time when the harvester is dragging in [...]

Yes exactly, the harvester is not docked to the refinery when the capturing happens.
